### ISSUE TITLE
Add `singleValue` option to `object-curly-spacing` rule

### ISSUE THREAD

**Issue by rasenplanscher** [eslint/eslint#8045]
I'd like to add to the `object-curly-spacing` rule a new option `singleValue` that would work analogous to the same option in `array-bracket-spacing`, like so:
:-1:  Examples of **incorrect** code for this rule with the `"always", { "singleValue": false }` options:
```js
/*eslint object-curly-spacing: ["error", "always", { "singleValue": false }]*/

var foo = { foo: 'foo' };
var foo = { foo: 'foo'};
var foo = {foo: 'foo' };
var foo = { n: 1 };
var foo = { n: 1};
var foo = {n: 1 };
var foo = { [ 1, 2 ] };
var foo = { { foo: 'foo', bar: 'bar' } };
var foo = { { 
  foo: 'foo', 
  bar: 'bar' 
} };
```

:+1:  Examples of **correct** code for this rule with the `"always", { "singleValue": false }` options:
```js
/*eslint object-curly-spacing: ["error", "always", { "singleValue": false }]*/

var foo = {foo: 'foo'};
var foo = {n: 1};
var foo = {[ 1, 1 ]};
var foo = {{ foo: 'foo', bar: 'bar' }};
var foo = {{ 
  foo: 'foo', 
  bar: 'bar' 
}};
```

**Comment by platinumazure** [comment]
Hi @rasenplanscher, thanks for the issue.

Would you mind reviewing our [rule change proposal guidelines](http://eslint.org/docs/developer-guide/contributing/rule-changes) and ensuring your first post answers all the questions we ask in our [rule change proposal template](https://github.com/eslint/eslint/blob/master/templates/rule-change-proposal.md)? You've answered a few of them already but it's much easier for us to review these proposals if they follow the same basic template. Thanks!

**Comment by rasenplanscher** [comment]
Sorry for that, wrote it in kind of a hurry. Had some mistakes in there, consequently... I hope the following is to your liking 😄 

#### What rule do you want to change?
`object-curly-spacing`

#### Does this change cause the rule to produce more or fewer warnings?
It neither increases nor decreases the number of warnings, generally.

#### How will the change be implemented? (New option, new default behavior, etc.)?
New option: `{boolean} singleValue`

#### Please provide some example code that this change will affect:

:-1:  Examples of **incorrect** code for this rule with the `"always", { "singleValue": false }` options:
```js
/*eslint object-curly-spacing: ["error", "always", { "singleValue": false }]*/

var foo = { foo: 'foo' };
var foo = { foo: 'foo'};
var foo = {foo: 'foo' };
var foo = { n: 1 };
var foo = { n: 1};
var foo = {n: 1 };
var foo = { [ 1, 2 ] };
var foo = { { foo: 'foo', bar: 'bar' } };
var foo = { { 
  foo: 'foo', 
  bar: 'bar' 
} };
```

:+1:  Examples of **correct** code for this rule with the `"always", { "singleValue": false }` options:
```js
/*eslint object-curly-spacing: ["error", "always", { "singleValue": false }]*/

var foo = {foo: 'foo'};
var foo = {n: 1};
var foo = {arr: [ 1, 1 ]};
var foo = {obj: { foo: 'foo', bar: 'bar' }};
var foo = {obj: { 
  foo: 'foo', 
  bar: 'bar' 
}};
```

#### What does the rule currently do for this code?
Currently, the desired behaviour can be effected partially using `arraysInObjects` and `objectsInObjects`. They fall short in that they only work for the brackets/braces, not the keys, and that they do not affect non-array and non-object values.

#### What will the rule do after it's changed?
The option allows switching the basic rule for the specific case that an object contains only a single value.

**Comment by platinumazure** [comment]
Thanks! I've left a :+1: on the original issue post. If a core team member is willing to champion this enhancement and two more people leave :+1:s, then we can accept the issue.

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get enough support from the team and so I'm closing it. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to reach consensus after a long time tend to never do it, and as such, we close those issues. This doesn't mean the idea isn't interesting, just that it's not something the team can commit to.
